bin/truffleruby

```diff
@@ -120,7 +138,28 @@ if [ -n "$CP" ]; then
 fi
 
 declare -a full_command
-full_command=("$JAVACMD" $JAVA_OPTS "-Xbootclasspath/a:$boot_cp_jar" "${java_args[@]}" org.truffleruby.Main "-Xhome=$root" "-Xlauncher=$root/bin/truffleruby" "${ruby_args[@]}")
+if [ $on_graalvm = false ]; then
+    full_command=(
+        "$JAVACMD"
+        $JAVA_OPTS
+        "-Xbootclasspath/a:$boot_cp_jar"
+        "${java_args[@]}"
+        org.truffleruby.Main
+        "-Xhome=$root"
+        "-Xlauncher=$root/bin/truffleruby"
+        "${ruby_args[@]}"
+    )
+else
+    full_command=(
+        $JAVACMD # no " to let options expand to multiple array elements
+        $JAVA_OPTS
+        "${java_args[@]}"
+        com.oracle.graalvm.Main
+        org.truffleruby.Main
+        "-Xlauncher=$root/bin/truffleruby"
+        "${ruby_args[@]}"
+    )
+fi
```

Comment:
There is almost no shared code (just `-J-cmd` and parsing simple options) beside the symlink resolving which needs to happen in all GraalVM launchers anyway.
I think it's not worth the extra complexity. One could make a shared bash file for shared functions if duplication of the flags parsing is so important.